src/com/belsoft/project/CustomDictionary.java

```diff
@@ -0,0 +1,64 @@
+package com.belsoft.project;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CustomDictionary {
+
+    private final Map<String, String> names = new HashMap<>();
+    private final Map<String, String> ports = new HashMap<>();
+    private final Map<String, String> ips = new HashMap<>();
+
+    public void add(String key, String value) {
+        if (PortValidator.isValid(key)) {
+            ports.put(key, value);
+        }
+        else if (IpAddressValidator.isValid(key)) {
+            ips.put(key, value);
+        }
+        else {
+            names.put(key, value);
+        }
+    }
+
+    public void printAll() {
+        names.keySet().stream()
```

Comment:
We have code duplication here, 3 chunks of code that basically do the same thing. This should be at least extracted in a separate method (for example, public void printAll(Map<String, String> map)), that could be independently unit tested.